cartridge/invalid-format.lua

```diff
@@ -0,0 +1,68 @@
+local log = require('log')
+
+local illegal_types = {
+    [''] = true,
+    ['n'] = true,
+    ['nu'] = true,
+    ['s'] = true,
+    ['st'] = true,
+}
+
+rawset(_G, '__cartridge_invalid_format_spaces', {})
+
+-- _space trigger.
+local function before_replace(_, tuple)
+    if tuple == nil then return tuple end
+
+    local name = tuple[3]
+    local format = tuple[7]
+
+    for _, field_def in ipairs(format) do
+        if illegal_types[field_def.type] or illegal_types[field_def[2]] then
+            _G.__cartridge_invalid_format_spaces[name] = true
+        end
+    end
+
+    return tuple
+end
+
+-- on_schema_init trigger to set before_replace().
+local function on_schema_init()
+    box.space._space:before_replace(before_replace)
+end
+
+local function spaces_name_to_str()
+    local res = ''
+    for name, _ in pairs(_G.__cartridge_invalid_format_spaces) do
+        res = res .. name .. ', '
+    end
+    return res:sub(1, -3)
+end
+
+return {
+    start_check = function()
+        if box.ctl.on_schema_init == nil then
+            return
+        end
+        box.ctl.on_schema_init(on_schema_init)
+    end,
+    end_check = function()
+        if box.ctl.on_schema_init == nil then
+            return
+        end
+        box.space._space:before_replace(nil, before_replace)
+        box.ctl.on_schema_init(nil, on_schema_init)
+
+        if next(_G.__cartridge_invalid_format_spaces) then
+            log.warn(
```

Comment:
The issue (#1985) asks to migrate the bad formats. Is there implecit agreement to issue a warning instead?

BTW, the issue will not be shown on tarantool 2.10.4. We'll get a fail on bootstrap. Just noted. Don't know, whether it is intentional.